### ISSUE TITLE
Unhelpful error in dynamic keyword argument interpolation

### ISSUE THREAD

**Issue by AA-Turner** [PyCQA/pylint#10822]
### Bug description

Consider:

```python
"""..."""


def test_symbol_coverage(thing):
    """..."""
    amount = 7
    dynamic_kwarg = {f"{thing}_amount": amount}
    func_taking_keyword_arguments(**dynamic_kwarg)


def func_taking_keyword_arguments(lobster_amount=0, spam_amount=0):
    """..."""
    del lobster_amount
    del spam_amount
```

Running `pylint bug.py` yields

```text
************* Module bug
bug.py:8:4: E1123: Unexpected keyword argument 'Uninferable_amount' in function call (unexpected-keyword-arg)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 2.86/10, +0.00)
```

Pylint appears to be failing to infer the value of `thing`, but then interpolating it into the format string, serialising the uninferable thing `thing` as the literal string `Uninferable`. This is then taken with the other text in the format string literal and matched against the parameters of the given function, which fails.

I think pylint should do nothing here, stopping flow analysis as soon as it encounters an uninferable variable.

### Configuration

```ini

```

### Command used

```shell
pylint bug.py
```

### Pylint output

```python
************* Module bug
bug.py:8:4: E1123: Unexpected keyword argument 'Uninferable_amount' in function call (unexpected-keyword-arg)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 2.86/10, +0.00)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 4.0.4
astroid 4.0.3
Python 3.14.2 (main, Dec  5 2025, 16:49:16) [Clang 17.0.0 (clang-1700.4.4.1)]
```

### OS / Environment

macOS Tahoe 26

### Additional dependencies

```python

```
